data-processing-lib/spark/src/data_processing_spark/runtime/spark/spark_transform.py

```diff
@@ -0,0 +1,12 @@
+from typing import Any
+
+from data_processing.transform.abstract_transform import DATA, AbstractTransform
+from pyspark.sql import DataFrame
+
+
+class AbstractSparkTransform(AbstractTransform[DataFrame]):  # todo: Remove double quotes
+    def __init__(self, config: dict):
+        self.config = config
+
+    def transform(self, data: DataFrame) -> tuple[list[DataFrame], dict[str, Any]]:
+        raise NotImplemented()
```

Comment:
This is fundamentally different from Python/Ray. Just bite a bullet and create a separate AbstractSparkTransform. Adding here AbstractTransform does not really add anything to the code. Its OO for the sake of OO nothing else. Also by using DataFrames here we are effectively limiting input to the Arrow tables. Just making sure that we understand this. There is no way back to the binary blobs

Reply:
This comes in because of the AbstractTransformLauncher -> runtime config -> transform config -> AbstractTransform.  Spark code is using the AbstractTransformLauncher and therefore we have a common AbstractTransform across all runtimes/launchers.

Reply:
complexity is in the ey of the beholder :)